tests/web/test_websocket_server.py

```diff
@@ -154,15 +156,16 @@ def test_websocket_string_event(test_client, loop):
 
 
 @asyncio.coroutine
-def test_websocket_block_event(b, test_client, loop):
+@pytest.mark.parametrize('_block', (10,), indirect=('_block',), ids=('block',))
+def test_websocket_block_event(b, _block, test_client, loop):
     from bigchaindb import events
     from bigchaindb.web.websocket_server import init_app, POISON_PILL, EVENTS_ENDPOINT
 
     event_source = asyncio.Queue(loop=loop)
     app = init_app(event_source, loop=loop)
     client = yield from test_client(app)
     ws = yield from client.ws_connect(EVENTS_ENDPOINT)
-    block = create_block(b, 10).to_dict()
+    block = _block.to_dict()
```

Comment:
Yes, I understand. 

For this specific case if we make the default number of transactions in a block to be `10` then the decorator part is not needed. Moreover the fixture could simplified, such that we end up with something like:

```python
@pytest.fixture
def block(b, request):
    total = getattr(request, 'param', 10)
    # ...

def test_websocket_block_event(b, block, test_client, loop):
    # ...
    block_dict = block.to_dict()
    # ...
```

The whole point with this approach is that it provides us with a fixture we can re-use, and when we need to fine tune the quantity of transactions that the block contains we can do so with [pytest's `indirect` parametrization](https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments) mechanism.

I think that the complexity is worth it in this case given what it provides us with, but that is my opinion. Lastly, tests in which this pattern is encountered is not un-common.